Support multiple SQL queries in Dataproc SQL job

amirmor1 opened the pull request:
<!-- Please keep an empty line above the dashes. -->

DataProcJobBuilder.add_query(query) is misleading, because it can make you think that you can call this function multiple times with different queries and then execute and it will send all queries, but it fact it will sent the last one since its override the queries.

I've added set_queries function which takes a list of strings and send it. Dataproc supports queries list.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

Commit:
Send multiple queries on dataproc sql job

DataProcJobBuilder.add_query(query) is misleading, cause it can make you think that you can call this function multiple times with different queries and then execute and it will send all queries, but it fact it will sent the last one since its override the queries.

I've added set_queries function which takes a list of strings and send it.
Dataproc supports queries list.

Commit:
Send multiple queries on dataproc sql job

DataProcJobBuilder.add_query(query) is misleading, cause it can make you think that you can call this function multiple times with different queries and then execute and it will send all queries, but it fact it will sent the last one since its override the queries.

I've added set_queries function which takes a list of strings and send it.
Dataproc supports queries list.

shahar1 commented on providers/src/airflow/providers/google/cloud/hooks/dataproc.py:
```diff
@@ -124,6 +124,14 @@ def add_query(self, query: str) -> None:
+
+        :param queries: queries for the job.
+        """
+        self.job["job"][self.job_type]["query_list"] = {"queries": queries}
```
Instead of creating another method, I think that it would be nicer to merge it into the existing `add_query`

amirmor1 replied:
Agree, updated.

Commit:
Append query to the queries sent to dataproc instead of replacing

Requested by reviewer, instead of adding a function that sets a list of queries, just fix the original function add_query that will actually append query to the queries list send to dataproc job.

Commit:
Append query to the queries sent to dataproc instead of replacing

Requested by reviewer, instead of adding a function that sets a list of queries, just fix the original function add_query that will actually append query to the queries list send to dataproc job.

shahar1 commented on providers/src/airflow/providers/google/cloud/hooks/dataproc.py:
```diff
@@ -118,11 +118,11 @@ def add_args(self, args: list[str] | None = None) -> None:
 
     def add_query(self, query: str) -> None:
```
Don't you want maybe to allow a list of queries (`query: str | list[str]`)? it will be nicer to handle rather than repeating:
```python
hook.add_query("query1")
hook.add_query("query2")
.
.
.
```

Also, could you please merge/rebase `main`?

amirmor1 replied:
agree, updated

Commit:
formatter

Commit:
Merge branch 'main' of github.com:apache/airflow into support-multiple-queries-in-dataproc

Commit:
merge master
